Mathlib/Order/SuccPred/Basic.lean

```diff
@@ -505,18 +505,18 @@ instance [PartialOrder α] : Subsingleton (SuccOrder α) :=
     · exact (@IsMax.succ_eq _ _ h₀ _ ha).trans ha.succ_eq.symm
     · exact @CovBy.succ_eq _ _ h₀ _ _ (covBy_succ_of_not_isMax ha)⟩
 
-section CompleteLattice
-
-variable [CompleteLattice α] [SuccOrder α]
-
-theorem succ_eq_iInf (a : α) : succ a = ⨅ (b) (_ : a < b), b := by
-  refine le_antisymm (le_iInf fun b => le_iInf succ_le_of_lt) ?_
+theorem succ_eq_sInf [CompleteLattice α] [SuccOrder α] (a : α) :
+    succ a = sInf (Set.Ioi a) := by
+  apply (le_sInf fun b => succ_le_of_lt).antisymm
   obtain rfl | ha := eq_or_ne a ⊤
   · rw [succ_top]
     exact le_top
-  exact iInf₂_le _ (lt_succ_iff_ne_top.2 ha)
+  · exact sInf_le (lt_succ_iff_ne_top.2 ha)
 
-end CompleteLattice
+theorem succ_eq_csInf [ConditionallyCompleteLattice α] [SuccOrder α] [NoMaxOrder α] (a : α) :
```

Comment:
If `a` is maximal, then `succ ⊤ = ⊤ = sInf ∅ = sInf (Ioi ⊤) = sInf (Ioi a)`
````suggestion
theorem succ_eq_csInf [ConditionallyCompleteLattice α] [SuccOrder α] (a : α) :
```

Reply:
There's two theorems in that sequence I can't find
1. `IsMax a → IsTop a` for conditionally complete lattices
2. `IsMax a → sInf Ø = a`
Any idea if we have them? If not, I can add them in this PR.

Reply:
`IsMax.isTop`

Reply:
Actually, I don't believe 2 is provable. All lemmas in the definition of a conditionally complete lattice assume that the set is nonempty.

Reply:
Hmm yeah